### ISSUE TITLE
XP Gain?

### ISSUE THREAD

**Issue by random-person-001** [random-person-001/MagicText#17]
We should have enemies give you xp upon death.  Shouldn't be too hard, and default xp = their initial health.

**Comment by HazilTheNut** [comment]
And what would the xp do?

Our main form of progression in the game is the acquisition of magic
potatoes. Additionally, the armor and weapons you obtain also help form the
player's progression and growth in power.

So...what would it do?

On Fri, Sep 16, 2016 at 10:48 PM, Maniacal Chicken <notifications@github.com

**Comment by random-person-001** [comment]
Probably a +1 power boost for (damage?) spells every `n` xp you earn.  I mean, it would be pretty much the only motivation to kill everyday enemies you come across.

**Comment by HazilTheNut** [comment]
I do like the idea that there should be inherent value in attacking and
killing enemies, but experience systems scare me.

Here's my thinking:

Currently, the game's difficulty at any point in the game is easy to
derive, just imagine that the player has 70-80% of the game's items, which
gives you a sense of the power of the player's equipment and also how many
magic potatoes the player has. (20 + (Total Potatoes Collected \* (5/2))
generates an theoretically average player health and mana bar.

From there, you can infer how much health and damage enemies should have to
present a threat to the player. It's easy.

Experience makes all of that hard. Play-test-requiring hard. Because the
player's patience is variable, I have no idea how many enemies the player
has killed. Not only that, enemies are also infinite in quantity, because
rooms don't save what enemies have been slain. Since the player now has no
roof on how powerful he/she can be, the player can farm/grind out exp to
effectively win the game.

If we put too much exp, the game is too easy and anything harder than that
is quickly brought down due to the ability to grind.
If we put too little, the game requires grinding, which is not really that
fun.

Striking that balance is difficult and rigorous. Everything about it stands
on a knife's edge.

That's why I don't like it.

Experience-based progression is meant to add achievable and easily
quantifiable goals in a game, and as a side effect a feeling of growth of
power in the player. Our game is silly and somewhat aimless. The player
doesn't really have a goal. They just go where forward seems to be, and
that carries them to the end of the game. All the while having fun.

Simply put, our game is more about the journey than the end goal, and
adding exp would shift the player's focus on the goal, and thus away from
the real attraction of the game.

Having no exp points also encourages the player to look around in an area,
making the content we create more likely to be seen and interacted with.

Anyway, that's my...two cents? Sounds more like a dollar to me, though.

**Comment by random-person-001** [comment]
@zachllogan , do you have thoughts on this?

**Comment by zachllogan** [comment]
xp is a rather quintessential game mechanic, it is even sometimes used as a symbol to represent video games. xp is often one of the main ways of giving enjoyment to the player; they feel proud and rewarded by leveling up. Really one could have a game that has xp and leveling that doesn't do anything besides giving the player a new title, and there would still be people who want to level up. Thought there are of course plenty of games that don't have xp. These usually are:

```
1. Heavily story driven games. These games don't necessarily try to be games, but more interactive stories. They don't need xp because the player gets their enjoyment from the story. The game could be this(it seemed Jared was suggesting as much), but their then should be a VERY complete story.
2. Mechanic driven games. These don't need xp because the player gets their enjoyment from playing with, investigating, or mastering the games mechanic(s). right now your game doesn't really have any interesting or deep mechanics or combat, so this doesn't really apply (and I don't think it will ever apply that much?)
3. Games structured into levels or rouge likes or party games. These usually don't have xp because they are very serial, casual, or one-off sorts of games. The player gets their enjoyment from surpassing a challenge or social competition. This doesn't really apply to the game because it is more adventure/quest structured.
4. Games with other progression systems. These don't need xp because the player gets stronger/more skills through some other system. Though this really is the same as xp just under a different name, so one could say the game already has an xp system.
```

Though those characteristics are hardly mutually exclusive with xp; I think even a heavily storied game of this style (the game seems very rpg-esque, the perfect xp audience BTW) could benefit from an xp system (unless its basically a book, in which case one might worry about gameplay getting in the way of the story.) 

Like I said I think the game already has an xp system, which I think brings along most of the negatives Jared pointed out, except perhaps the grinding (though searching for potatoes, if they are hard to find, could be considered grinding), especially because you can't accurately predict how many potatoes a player will have collected. trust me, some people will collect every single one and some will only collect about 1/2 or even 1/3; the variation is probably higher than you think, unless the potatoes are really easy to find, which would be boring, or the game is really hard without collecting most of them (which again takes away from fun/ is grindy).

I do, however, think the game could benefit from the addition of a second, more sophisticated xp (or equivalent) system; perhaps a system that would allow the player to purchase new skills or upgrades to skills (skills could be of the player or of weapons/spells) (basically a skill tree). These skills would not be necessary to complete the game, but would make certain tasks a little easier (bare in mind that it doesn't necessarily have to make them easy per se, but perhaps more convenient; consider the player having an annoying, regular task to do that's not necessarily hard, but they would rather not have to waste their time on it, by allowing them to work to relieve themselves of this duty, they feel rewarded when they EARN no longer having to do it), would allow more possibilities for completing an area/task, would allow for more personalized play styles, or would allow access to new areas. This system would not require grinding, as progression through the game wouldn't depend on ever spending any skill points, though the system would require adding more challenge, variance, tasks, and/or depth to the game, which I think should be done anyways.

Also about the games difficulty and balancing, I don't think their should be much stress over it; the skill that players have for the tasks the game requires will vary a lot between individual players, and many games have multiple difficulty levels for that very reason, so unless they are way out of wack I don't think it is a very big deal. Furthermore their are ways of controlling and/or capping xp. Also, I think play testing is a good idea and not that difficult, hell I have seen crappy flash-games that have been play tested. 

But this is all just my opinion, and if the game is EXTREMELY story based then xp might not be necessary and might even take away from the story or if it had very deep and challenging tasks xp might also not be necessary; I just think these might not be the case and that an xp system might add a little bit more interest and motivation/reward to the game (as well as giving a reason to actually engage in combat.)
